train_npi.py

```diff
@@ -1066,93 +1062,102 @@ def train_adversarial_NPI(args): # train NPI and Classifiers in-tandem
             loop = tqdm(total=len(train_file_names), position=0, leave=False)
             for file_num, train_file_name in enumerate(train_file_names):
                 gc.collect()
-                train_data, test_data = load_training_data(train_file_path+train_file_name, args.perturbation_indices, split_ratio=.25, filter_unk=False, permitted_rows=None) # val_data, _ and .25
+                train_data, test_data = load_training_data(train_file_path + train_file_name, args.perturbation_indices,
+                                                           split_ratio=.25, filter_unk=False,
+                                                           permitted_rows=None)  # val_data, _ and .25
 
                 # CREATE TRAIN / TEST LOADERS
                 train_loader = NPIDataLoader(train_data, batch_size=batch_size, pin_memory=True)
                 test_loader = NPIDataLoader(test_data, batch_size=batch_size, pin_memory=True)
 
                 npi_batch_losses = []
                 generate_class_batch_losses = []
-                generate_class_train_batch_accuracies = []                
+                generate_class_train_batch_accuracies = []
 
                 # Looping through training batches
                 for batch, (orig_activ, real_label, target_label, data_inds) in enumerate(train_loader):
-                    
-                    functional_batch_size = orig_activ.shape[0] 
+
+                    functional_batch_size = orig_activ.shape[0]
 
                     # prepare the batch for model processing
                     input_activs_shape = orig_activ.size()
-                    input_targ_shape = (1,1) # target_label.size() <-- this is None
-                    orig_activ, real_label = orig_activ.cuda(async=True).float(), \
-                                             real_label.cuda(async=True).float()
+                    input_targ_shape = (1, 1)  # target_label.size() <-- this is None
+                    orig_activ, real_label = orig_activ.cuda(non_blocking=True).float(), real_label.cuda(
```

Comment:
Interesting edit. This part of our code must have been copied from another source, because neither of us actually knew what the `async` keyword is used for. I have never heard of `non_blocking` for that matter. If you can clarify at all the reason for this change, please do!